### PR TITLE
node-sass -> dart-sass

### PR DESCRIPTION
**Pull request by illusionfield**
Closing this PR in favor of the consolidated version here:  
➡️ **#329 – Finalize Dart Sass migration & drop direct dependency**

That PR targets `release/5.0.0` and includes all changes from this one, along with updated docs and migration instructions.  
Looking forward to your feedback there!

Hi,

I've spent quite some time working on the integration of Meteor JS and Dart Sass. Given how things are evolving with Dart Sass, I wanted to share my approach. I'm really curious to hear the feedback from the maintainer community on whether this direction is viable.

- It requires Meteor 3
- Dropped the `seba:minifiers-autoprefixer` dependency and aligned it with Meteor’s standard-minifier, which now generates perfect mappings.
- Reworked the `scss-config.json` for improved handling.
- The `includePaths` config option seems less relevant now, but this might need further review to cover any cases I might have missed.
- Due to unique import strategies in both Meteor and Dart Sass, I had to sync up the import functionality, so it’s not fully backward-compatible.  Dart Sass's importer returns a URL object, while its input is an URL string, which would have broken the `@import "{my-package:pretty-buttons}/...";` implementation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,132 +21,112 @@ Package.onUse(function (api) {
...
-users can opt into one by one.
+```javascript
+api.addFiles('x.scss', 'client', { isImport: true });
+```
 
-Each compiled source file produces a separate CSS file.  (The
-`standard-minifiers` package merges them into one file afterwards.)
+Each compiled source file generates a separate CSS file, which is then merged into one by the `standard-minifiers` package.
 
 ### Importing
```
Comment (StorytellerCZ) [inline README.md]: Looking at the changes here, this requires a migration guide to be included here.

**Review by StorytellerCZ — APPROVED** [review]
Just need more documentation on migration. Other than that I'm happy to move this forward.

File: package.js [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
-    'caching-compiler@2.0.0',
-    'ecmascript@0.16.9'
+    'caching-compiler@1.2.2 || 2.0.0',
+    'ecmascript@0.16.7 || 0.16.10',
   ],
   sources: ['plugin/compile-scss.js'],
   npmDependencies: {
-    "node-sass": '9.0.0',
-    "@babel/runtime": "7.24.5"
+    '@babel/runtime': '7.26.0',
```
Comment (jankapunkt) [inline package.js]: Is it necessary to pin this version? Isn't it installed on app-level anyway?

**Review by jankapunkt — APPROVED** [review]
I would definitely move forward with this but I would publish an alpha version first, so we can field test this with some real-world apps.

File: package.js (hunk 2 above)
Comment (illusionfield) [reply package.js]: You're absolutely right! However, without this, the TinyTest GUI crashes. 

It's only needed for that, and I agree that it's quite wasteful. The entire test setup should be modernized as well.

Comment (jankapunkt) [reply package.js]: You can define depnedbcies as test only by calling npm depends within onUse and onTest separately as long as there is no npm.depends outside of them

Comment (illusionfield) [reply package.js]: Thanks for the info, I checked it out!

We can actually drop it entirely because I couldn't get it to work with Meteor 2.8 anyway. The minimum supported Meteor version is now 2.10, and I tested it without the dependency there—everything worked fine, just like with Meteor 3.

**Comment by illusionfield** [comment]
The issue with Meteor 2.8 compatibility was that when I published the package with a Meteor 3.x release, it wasn’t compatible with `fibers` in a 2.8 installation. On the other hand, if I published it with a 2.8 release, it couldn’t find `fibers` in installations above 3.0.

This makes sense given the structural changes in Meteor 3, but I couldn’t find a real solution to make it work seamlessly across both versions. That’s why I set the minimum supported Meteor version to 2.10.

I found more details about this issue here:  
[https://forums.meteor.com/t/solved-is-it-possible-to-publish-a-meteor-compiler-package-that-works-for-both-2-0-and-3-0-yes](https://forums.meteor.com/t/solved-is-it-possible-to-publish-a-meteor-compiler-package-that-works-for-both-2-0-and-3-0-yes)

File: README.md (hunk 1 above)
Comment (illusionfield) [reply README.md]: Would something like this be a good starting point? 

[migration-guide.md](https://github.com/user-attachments/files/18708547/migration-guide.md)

**Comment by StorytellerCZ** [comment]
@jankapunkt now that `@babel/runtime` has been remove, I think we can move to releasing a beta and move this forward to full and proper release. We will ask people to test and give feedback again on Friday and then let's move to final release quickly.

**Comment by DblK** [comment]
I was looking to move to the next version to avoid issue with install of python for node-sass, but one key feature is missing `includePaths`.
Do you plan to add it back for final `5.0.0` version?

**Comment by illusionfield** [comment]
@DblK

This feature has theoretically lost its purpose in 5.0 since predefined search paths are no longer necessary for relative imports.

The `@import` (or now `@use`) works in multiple ways:
- Relative paths: `../../example`
- Project absolute path: `'{}/imports/...'`
- Node modules: `~pkg` (resolves to `project/node_modules/pkg`)
- Absolute file system paths: `"/home/user/dev/globalstyles"` or `"C:\Styles\fonts"`

However, if there are specific cases where `includePaths` is still needed, let’s discuss it. There might be some edge cases I haven’t considered.

**Comment by DblK** [comment]
I do understand there are some workaround and I can manage this, however I have one file called `_mixins.scss` and I have dozen of `scss` refering to it by `@import "mixins"` and my `includePaths` set to `['{}/client']`.
The only way to make it work is to update all the scss files to `@import "{}/client/mixins"` (or better use `@use` for better compatibility for the next version.)

I just wanted to avoid the use of `{}/client/` in every single `scss` file. `includePaths` was very helpful for that. But if there is no other way..

And I am lucky if my `includePaths` was `['{}/some/very/long/path/for/holding/my/mixins/folder']`, it would have been more painful to add it in every single scss file 😅.

**Comment by illusionfield** [comment]
I see your point, and I totally understand why `includePaths` was useful in that case.

A good alternative is using `@forward` and `@use`, which are the recommended approaches in modern Sass versions. Instead of relying on `includePaths`, you can create a central file (e.g., `client/mixins/_index.scss`) and forward your mixins like this:

```scss
@forward "{}/some/very/long/path/for/holding/my/mixins/folder";
```

Then, in your SCSS files, instead of @import "mixins", you can simply use:

```scss
@use "{}/client/mixins" as *;
````

This way, all your mixins and variables are available everywhere without having to update each file individually. It’s more modular and keeps your paths clean while ensuring better compatibility with future Sass versions.

I hope I understood your issue correctly, but if not, let’s discuss it further! 😊

Maybe it's worth considering a hot prefix for `"{}/client/"` and `"{}/imports/"`, similar to how `~` works for `"{}/node_modules/"`, as an alternative to `includePaths`.  

@StorytellerCZ, what do you think about this?  

Whichever direction we take, I believe this feature discussion should be moved to a separate issue.

**Comment by StorytellerCZ** [comment]
Don't really have opinion here. Since we are doing a major version upgrade it is allowed to break some things, just need to ensure that it is properly documented and mentioned in the migration docs.
If it is not much hassle easing the transition by this is fine with me, but then I would be for that we highlight it as such and that people shouldn't expect this to work in the next major release.

**Comment by dr-dimitru** [comment]
@illusionfield I agree with @StorytellerCZ that major bump will require detailed migration guidelines and/or references to "modern" SCSS/SASS practices, especially on including/importing files

**Comment by illusionfield** [comment]
@dr-dimitru 
I agree, and I've already started working on a draft version ([here](https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#discussion_r1946693456)). It looks like some additional adjustments will be needed, especially regarding the `includePaths` issue.  

I wouldn’t rule out the possibility of it being merged back soon, but there's also a critical bug that just came up ([here](https://github.com/Meteor-Community-Packages/meteor-scss/issues/324)), so there's still some work to do to ensure everything is in good shape for everyone.

File: README.md (hunk 1 above)
Comment (dr-dimitru) [reply README.md]: @illusionfield regarding migration guide:
- Add short description of difference between `@use` and `@forward` with reference to the docs
- Run mdlinter over the file (Bold fonts in headers are rudimentary)
- The rest looks good 👍 let's attach it to this PR

File: README.md [hunk 3]
```diff
@@ -25,132 +21,112 @@ Package.onUse(function (api) {
...
-```
+You can enable debug mode by setting the `DEBUG_PACKAGE_SASS` environment variable to any value other than `false` or `0`. In debug mode, additional error messages and stack traces will be logged to the console.
 
-In a Meteor 1.3+ project, do the same by running:
-```
-meteor remove standard-minifier-css
-meteor add seba:minifiers-autoprefixer
+```bash
+export DEBUG_PACKAGE_SASS=true
 ```
```
Comment (dr-dimitru) [inline README.md:176]: @illusionfield I'd suggest keeping short description about prefixer in the main Readme file, for example as [in my PR](https://github.com/Meteor-Community-Packages/meteor-scss/pull/325/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R154-R162)

**Comment by dr-dimitru** [comment]
@illusionfield I left comments for migration doc and changes in readme

File: README.md [hunk 4]
```diff
@@ -25,132 +21,112 @@ Package.onUse(function (api) {
...
-</thead>
-<tbody>
-<tr><td>1.0 - 1.1</td><td>3.2.0</td></tr>
-<tr><td>1.2 - 1.3.1</td><td>3.4.2</td></tr>
-<tr><td>1.3.2+</td><td>3.8.0_1</td></tr>
-<tr><td>1.4.0</td><td>3.8.1</td></tr>
-<tr><td>1.4.1+</td><td>4.5.4</td></tr>
-<tr><td>1.6+</td><td>4.12.0</td></tr>
-</tbody>
-</table>
```
Comment (dr-dimitru) [inline README.md:40]: @illusionfield I'm suggesting to keep this table for historical reference and to provide support for older projects

**Review by dr-dimitru — CHANGES_REQUESTED** [review]
Please see my comments regarding documentation

**Comment by illusionfield** [comment]
### Updates in this PR

- The `includePaths` option is now available again. (@DblK)
- Fixed issue #324.
- The **Migration Guide** has been completed.

### Help Needed

@dr-dimitru, @StorytellerCZ

I would appreciate some help with finalizing the **README**.  
I'm unsure about what key points to highlight and how to effectively focus on the features introduced with this update without repeating content already covered in the **Migration Guide**.

Any suggestions or feedback would be greatly appreciated!

File: README.md (hunk 4 above)
Comment (illusionfield) [reply README.md:40]: @dr-dimitru I have restored it as you suggested.

**Comment by dr-dimitru** [comment]
@illusionfield good, will review shortly. In general it's okay to repeat some key things, where migration guide will be very detailed document

File: MIGRATION-GUIDE.md
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md:1]: I'm suggesting to add version name to Migration-guide file-name

File: MIGRATION-GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+```sh
+meteor update fourseven:scss
+```
+
+This automatically switches to the Dart Sass-based compiler.
+
+**Important:** The minimum required Meteor version is 2.10, and the package fully supports Meteor 3.
+
+### 🚨 Critical: Remove Seba Minifiers and Use Standard Minifier CSS 🚨
```
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md]: Change title to "Critical: Remove outdated packages"

File: MIGRATION-GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+## Migration Steps
+
+### Update the `fourseven:scss` package
+
+To use the latest version, update the package to the newest release:
+
+```sh
+meteor update fourseven:scss
+```
```
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md]: Add removal instructions right to the first section, as most users will quit after "update package"

```sh
meteor remove seba:minifiers-autoprefixer
meteor add standard-minifier-css
```

File: MIGRATION-GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+## Why is this change happening?
+
+- The Node.js-based `node-sass` relies on the outdated **LibSass** compiler, which is no longer maintained.
+- Dart Sass is the official Sass implementation that supports the latest standards and features.
+- Meteor 3 compatibility requires modern packages.
+- This transition ensures **better compatibility and long-term maintainability**.
+
+## Migration Steps
+
```
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md]: Add ToC between these two headers to indicate this is multi-step procedure:

- Step 1: Update...
- Step 2: remove outdated...
- Step 3: Install and use NPM packages for....
- Step 4: Change `@imports` to `@use`
- Step 5: Update variables usage

Each line can link to headers in this document

File: MIGRATION-GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+The `fourseven:scss` package **no longer uses `seba:minifiers-autoprefixer`**.
+Instead, it relies on **`standard-minifier-css`**, which includes built-in **autoprefixer** support.
+
+Failure to remove `seba:minifiers-autoprefixer` may lead to errors or incorrect CSS processing by the Meteor build system!
+
+```sh
+meteor remove seba:minifiers-autoprefixer
+meteor add standard-minifier-css
+meteor npm install --save-dev autoprefixer postcss postcss-load-config
```
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md]: Installing and configuring new autoprefixer should have separate section

File: MIGRATION-GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+## Additional Changes
+
+### Indented Syntax Support
+
+Previously, due to a limitation in LibSass, indented syntax (`.sass`) files could only import at the top level. This limitation has been removed in Dart Sass, allowing imports at any level.
+
+### Configuration File Update
+
+The configuration file has been renamed from `scss-config.json` to `.scss.config.json` to support a wider range of Dart Sass-specific settings.
```
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md]: Propose or reference recommended config file

File: MIGRATION-GUIDE.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+| Import Syntax          | `@import`                               | `@use` and `@forward`                    |
+| Module Import          | `@import "{}/node_modules/module-name"` | `@use "~module-name"`                    |
+| Meteor Package Import  | `@import "{package}/styles"`            | `@use "meteor:{package}/styles"`         |
+| Variable & Mixin Usage | Direct variable access                  | Prefix required (using `as *` or `as v`) |
+| CSS Minification       | `seba:minifiers-autoprefixer`           | `standard-minifier-css`                  |
+| Indented Syntax        | No import support                       | Fully supported                          |
+| Config File Name       | `scss-config.json`                      | `.scss.config.json`                      |
+| Debug Mode             | Not available                           | Enabled via environment variable         |
+
+This guide helps developers transition to the Dart Sass-based `fourseven:scss` package, ensuring compatibility with Meteor 3 and modern Sass features.
```
Comment (dr-dimitru) [inline MIGRATION-GUIDE.md]: Add "Further reading" section at the end of the document with:
1. Sum up of all used external links in the document
2. Links to mentioned packages
3. Related links where devs can learn more

File: README.md [hunk 11]
```diff
@@ -1,20 +1,20 @@
...
-This is a Sass build plugin for Meteor. It compiles Sass files with node-sass.
+# Dart Sass for Meteor.js
 
-**Note that due to a limitation in libsass, there is no support for imports with indented syntax (sass). Indented syntax does work on the top-level. A version based on dart-sass is in the works which should remove this limitation.**
+## Important Notice
 
-**Meteor 1.7 introduced a change in how node_modules are handled, if you want to import sass from a node_module you need to symlink the package in your imports directory (more information below)**
+This is a major change—from LibSass to Dart Sass. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding.
+
+This is a build plugin for Meteor.js that compiles Sass files using Dart Sass.
```
Comment (dr-dimitru) [inline README.md]: Let's keep this line right after opening `<h1>` header "Dart Sass for Meteor.js"

File: README.md [hunk 12]
```diff
@@ -1,20 +1,20 @@
-# Sass for Meteor
-This is a Sass build plugin for Meteor. It compiles Sass files with node-sass.
+# Dart Sass for Meteor.js
 
-**Note that due to a limitation in libsass, there is no support for imports with indented syntax (sass). Indented syntax does work on the top-level. A version based on dart-sass is in the works which should remove this limitation.**
+## Important Notice
 
-**Meteor 1.7 introduced a change in how node_modules are handled, if you want to import sass from a node_module you need to symlink the package in your imports directory (more information below)**
+This is a major change—from LibSass to Dart Sass. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding.
```
Comment (dr-dimitru) [inline README.md]: Let's use `[!IMPORTANT]` quote style of GitHub, I updated text a little as well:

> [!IMPORTANT]  
> In `v5.0.0` this package has introduced a major change — migrating underlying library `LibSass` to `Dart Sass`. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding

Markup:

```markdown
> [!IMPORTANT]  
> In `v5.0.0` this package has introduced a major change — migrating underlying library `LibSass` to `Dart Sass`. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding
```

File: README.md [hunk 13]
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
...
-`standard-minifiers` package merges them into one file afterwards.)
+Each compiled source file generates a separate CSS file, which is then merged into one by the `standard-minifiers` package.
 
 ### Importing
 
-You can use the regular `@import` syntax to import any Sass files: sources or
-imports.
+You can import styles from various locations, including other packages, your own app, and npm modules.
+
+- **Importing from another package**:
```
Comment (dr-dimitru) [inline README.md]: Better to use 4th level header here `####`

File: README.md [hunk 14]
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
...
 
 .my-button {
-  // use the styles imported from a package
-  @extend .pretty-button;
+  @extend .pretty-button; // No namespace required
 }
 ```
 
-Importing styles from the target app:
+- **Importing from your app**:
```
Comment (dr-dimitru) [inline README.md]: Better to use 4th level header here `####`

File: README.md [hunk 15]
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
...
 
 .my-nav {
-  // use a color from the app style pallete
-  background-color: @primary-branding-color;
+  background-color: @primary-branding-color; // Use a color from the app's style palette
 }
 ```
 
-This can also conveniently be used to import styles from npm modules for example:
+- **Importing from npm modules**:
```
Comment (dr-dimitru) [inline README.md]: Better to use 4th level header here `####`

File: README.md [hunk 16]
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
...
+
+### Autoprefixer
+
+To use Autoprefixer, follow the official [PostCSS with Meteor](https://docs.meteor.com/packages/standard-minifier-css.html#standard-minifier-css) documentation.
+
+## Custom Configuration
+
+You can customize the behavior of the Sass compiler by creating a `.scss.config.json` or `scss-config.json` file in the root of your project. This file supports options like `quietDeps`, `verbose`, and others that control Sass compilation behavior.
+
+Example configuration:
```
Comment (dr-dimitru) [inline README.md]: Is there reference to config file on Sass/Dart Sass resources that we can link here?

File: README.md [hunk 17]
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
...
-Sometimes a 3rd party module uses import paths that assume that the compiler is
-configured with specific `includePaths` option (e.g. Ionic, Bootstrap, etc.):
-```scss
-@import "ionicons-icons"; // This file is actually placed in another npm module!
-```
+Source maps are enabled by default, helping with debugging by mapping the compiled CSS back to the original Sass files.
+
+### Autoprefixer
+
+To use Autoprefixer, follow the official [PostCSS with Meteor](https://docs.meteor.com/packages/standard-minifier-css.html#standard-minifier-css) documentation.
```
Comment (dr-dimitru) [inline README.md]: I'd add autoprefixer installation here as Meteor's article focused more on PostCSS

```sh
meteor npm install --save-dev autoprefixer postcss postcss-load-config
```

The rest can be learned from PostCSS Meteor docs article

File: README.md [hunk 18]
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
...
 
-```
-meteor npm install the-package
-cd imports
-ln -s ../node_modules/the-package .
-```
+### Global Include Path
 
-See the [meteor changelog](https://github.com/meteor/meteor/blob/devel/History.md) for more information.
+At the moment, there is no support for a global include path. If there is significant demand, this feature could be added in future updates.
```
Comment (dr-dimitru) [inline README.md]: Isn't `includePaths` with `@forward` acts as global import?

**Review by dr-dimitru — CHANGES_REQUESTED** [review]
@illusionfield I left comments regarding docs. Please review, — I'm open for discussion

**Comment by illusionfield** [comment]
> @illusionfield I left comments regarding docs. Please review, — I'm open for discussion

@dr-dimitru Thank you very much, it's awesome! 

However, the README is in serious need of refactoring, and I'm afraid I might be short on time due to my deadline-driven work commitments. As a result, this PR will likely remain in a pending state, even though it forms the foundation of all my projects. :)

Another observation is that, since the package was almost entirely rewritten from scratch, the tests have become quite superficial. They do not cover many expected and unexpected edge cases, which makes even the smallest modifications require extensive manual testing. Keeping this in mind, there is still a lot of work ahead before the release. 

I appreciate any help in this process!

**Comment by jankapunkt** [comment]
Any demand for another package release? Are we still in alpha or is this already beta stage?

**Comment by illusionfield** [comment]
@jankapunkt 
Yes, we definitely need another alpha release since `includePaths` has been reintroduced and issue #324 has been fixed.  

I see the need for the community to test it, but unfortunately, the `README.md` is in a pretty bad state.

The code itself is in beta state—every feature has been implemented and tested. We are only dependent on the documentation, which is still in an early stage.

**Comment by dr-dimitru** [comment]
@illusionfield I can take on docs. Let me know where I can push my changes; @StorytellerCZ @jankapunkt or perhaps we can merge this PR as it is into separate branch (like `feature/dart-sass/5.0.0`) and continue working on it

**Comment by illusionfield** [comment]
Thank you so much, @dr-dimitru! 

This is really a great help. Feel free to reach out if you have any questions about how the code works.  

A `release/5` branch has been created from this PR—should I set the PR target to it?  

It’s also very important to update the package tests, and I’d be happy to put some effort into that.

File: README.md (hunk 18 above)
Comment (illusionfield) [reply README.md]: Currently, it seems necessary to support the `includePaths` option because, without it, the transition would place an even greater burden on users, who already face a significant effort in adapting their projects.  

Additionally, some level of support for the [Sass migrator](https://sass-lang.com/documentation/cli/migrator/) might still be required to facilitate the migration process.

**Comment by jankapunkt** [comment]
released 5.0.0-alpha.2 but accidentally pushed the commit to `dat-css`, please merge that branch, I will delete it once we have merged everything into Master at the very end.

**Comment by illusionfield** [comment]
@jankapunkt  I have successfully merged the dat-css branch into my branch and also bumped the version to 5.0.0-alpha.3.

**Comment by illusionfield** [comment]
Issue #326 has been raised about the migration guide not being accessible in the package registry. Since this is an important part of the transition, we should ensure it's correctly linked. What is the best practice for referencing a non-README file within an ongoing PR?

@dr-dimitru @StorytellerCZ @jankapunkt

**Comment by dr-dimitru** [comment]
@illusionfield use this link to the guide on your Fork — https://github.com/illusionfield/fourseven-scss/blob/master/MIGRATION-GUIDE.md

**Comment by dr-dimitru** [comment]
> Thank you so much, @dr-dimitru!
> 
> This is really a great help. Feel free to reach out if you have any questions about how the code works.
> 
> A `release/5` branch has been created from this PR—should I set the PR target to it?
> 
> It’s also very important to update the package tests, and I’d be happy to put some effort into that.

Any chance to get write permissions to your fork to manage changes to the docs?

**Comment by illusionfield** [comment]
@dr-dimitru 
Sorry for the delay. I have just sent the invitation!

**Comment by dr-dimitru** [comment]
> @dr-dimitru Sorry for the delay. I have just sent the invitation!

Well received, thanks! I'll try to propose changes this week

**Comment by illusionfield** [comment]
I (hopefully) fixed the issue #327 as well. Dart Sass has a significantly different path resolver ([Dart Sass Path Resolver](https://github.com/dart-lang/core/blob/0b2bd3fcd7f3e082f4cc9b14c19ffa93894b85ae/pkgs/path/lib/path.dart#L474)) compared to the Meteor package API's path handling ([Meteor Package API](https://github.com/meteor/meteor/blob/devel/docs/source/api/packagejs.md#accessing-file-system)).

I concluded that it makes sense to merge the two and utilize Dart Sass's path resolver, as it significantly improves the plugin's speed and expands the possibilities of path resolving without needing to implement each option separately.

This is yet another step towards the beta version.

@dr-dimitru If you can clean up the documentation chaos I created, it would be worth releasing the next version.

@jankapunkt Currently, the version is 5.0.0-alpha.3, but once the documentation is ready, it would make sense to release it as 5.0.0-beta.1.

**Comment by illusionfield** [comment]
[Here](https://github.com/Meteor-Community-Packages/meteor-scss/issues/327#issuecomment-2686019067)
a feature was mentioned that isn't well documented but seems interesting. It might be worth extending the documentation to cover this as well.

@dr-dimitru , @StorytellerCZ  what do you think?

**Comment by jankapunkt** [comment]
@illusionfield just let me know when there is a beta release to be published, I will then review and publish.

**Comment by illusionfield** [comment]
> @illusionfield just let me know when there is a beta release to be published, I will then review and publish.

@dr-dimitru when do you think the documentation is likely to be ready?
Thanks again for your help! 🙏🙏🙏

File: README.md (hunk 16 above)
Comment (illusionfield) [reply README.md]: yes, here:
[SASS Compile Options](https://sass-lang.com/documentation/js-api/interfaces/options/)
but currently only these 2 options are implemented. [See here](https://github.com/illusionfield/fourseven-scss/blob/092e7dba0bad192d328bcf7866fc760b35d1f168/plugin/compile-scss.js#L186)

**Comment by illusionfield** [comment]
I have updated the documentation—it's not necessarily perfect, but should be a good improvement. I have also fixed all issues related to the 5.0 release. The current version is **5.0.0-alpha.3**.

@jankapunkt I believe the package is ready for the beta release.

**Comment by jankapunkt** [comment]
I released `5.0.0-alpha.3` as this version was set in `package.js` but there was no release for it. Wen treat it like a beat though :-D

**Comment by illusionfield** [comment]
I hope the PR can be merged soon, as I would like to switch back from the temporary alternative to the official version in my projects. 😃

**Comment by illusionfield** [comment]
I took some inspiration from here: https://github.com/veliovgroup/flow-router/pull/123  
Hopefully, this will only make the package even stronger.

**Comment by illusionfield** [comment]
Hi everyone,  

Some time has passed, and the latest release seems to be working fine.  

@dr-dimitru, @StorytellerCZ Do you have any feedback or observations regarding the documentation?  

@jankapunkt If the documentation is also in good shape, then maybe this could be **rc.0** instead of beta.  
What would be the next steps to get this PR merged into the main branch as soon as possible?  

Additionally, I have a question:  
I have an advanced development version that no longer includes Sass as an npm dependency but allows it to be installed separately in `node_modules`. This is important because I use [Dart Sass Embedded](https://www.npmjs.com/package/sass-embedded) in my projects.
How would it be best to integrate this as an official option so it can evolve alongside this package if needed?

**Comment by jankapunkt** [comment]
> @jankapunkt If the documentation is also in good shape, then maybe this could be rc.0 instead of beta.

no problem

>  What would be the next steps to get this PR merged into the main branch as soon as possible?

We first need to make sure we have at least one or two more people testing this as part of their running apps. I have two mid-sized apps with bootstrap and some custom SCSS and would like to check this out. However, I'm currently occupied with some other things at work. 
Therefore I kindly ask @dr-dimitru and @StorytellerCZ to do some testing with their apps, if possible.
I just want to avoid this getting merged too soon and then incoming issue reports building immediate pressure (which I will not be able to handle then)

> Additionally, I have a question: I have an advanced development version that no longer includes Sass as an npm dependency but allows it to be installed separately in node_modules. This is important because I use [Dart Sass Embedded](https://www.npmjs.com/package/sass-embedded) in my projects. How would it be best to integrate this as an official option so it can evolve alongside this package if needed?

How is this integrated within your project? Simply by installing the package or by some "linking"? If you need to link this somewhere then I would go for a dependency-injection.

**Comment by illusionfield** [comment]
@jankapunkt 
I must confess—I "borrowed" the idea from an official Meteor package 😄  
[standard-minifier-css](https://github.com/meteor/meteor/blob/2d9cd4f1a06163f4c9d40abf9e317d3850fb2b2f/packages/standard-minifier-css/plugin/postcss.js#L29).  

Right now, it's still for [private use](https://github.com/illusionfield/meteor-dart-sass-development), and I'm evaluating whether it makes sense to proceed in this direction.  

I'm already using it in a few projects, and it works—but of course, that doesn't necessarily mean it's suitable for the official community package. That still needs to be determined.

**Comment by dr-dimitru** [comment]
> > @jankapunkt If the documentation is also in good shape, then maybe this could be rc.0 instead of beta.
> 
> no problem

I'll review/push changes sometime soon. Wrapping up now with my packages that await for update by community

> > What would be the next steps to get this PR merged into the main branch as soon as possible?
> 
> We first need to make sure we have at least one or two more people testing this as part of their running apps. I have two mid-sized apps with bootstrap and some custom SCSS and would like to check this out. However, I'm currently occupied with some other things at work. Therefore I kindly ask @dr-dimitru and @StorytellerCZ to do some testing with their apps, if possible. I just want to avoid this getting merged too soon and then incoming issue reports building immediate pressure (which I will not be able to handle then)

No problem, will test and report here soon

> > Additionally, I have a question: I have an advanced development version that no longer includes Sass as an npm dependency but allows it to be installed separately in node_modules. This is important because I use [Dart Sass Embedded](https://www.npmjs.com/package/sass-embedded) in my projects. How would it be best to integrate this as an official option so it can evolve alongside this package if needed?
> 
> How is this integrated within your project? Simply by installing the package or by some "linking"? If you need to link this somewhere then I would go for a dependency-injection.

@illusionfield Same question here, if it will be solved with simple `install development npm package X` in the docs — It's okay, if it's more than simply installing a package — I'd link it as dependency

**Comment by illusionfield** [comment]
@dr-dimitru 

The integration is as simple as running: 

```sh
meteor npm install –save-dev sass
```

or

```sh
meteor npm install –save-dev sass-embedded
```

No additional linking or special setup is required. 


## There are multiple reasons for this approach

- **Our company does not use a dedicated UI framework** (like Bootstrap, etc.), but instead, we have a custom framework built directly on raw Sass. Because of this, we pay extra attention to how Sass is handled in our workflow.

- **Dart Sass Embedded is significantly faster** than regular Dart Sass, but it is **not platform-independent**. Even `node-gyp` doesn’t solve this because it handles platform dependencies via `optionalDependencies` ([reference here](https://github.com/sass/embedded-host-node/blob/1ec3fbdf241267ce3c55eae4a4fb6d374ed370d9/package.json#L40C4-L40C24)).  
  This means bundling it into the Meteor build **is not feasible** since `publish-for-arch` is no longer available.

- The `quietDeps` option in Sass **only mutes warnings for dependencies** ([see docs](https://sass-lang.com/documentation/js-api/interfaces/options/#quietDeps)).
  - In version **1.79**, it did **not** include `@import` deprecation warnings.
  - However, in **^1.80**, `@import` deprecation warnings were introduced.
  - In our older projects where we haven’t yet migrated `@import` to `@use` or `@forward`, we **temporarily force** version **1.79** to **avoid excessive console warnings** while focusing on refactoring.

**Comment by illusionfield** [comment]
Closing this PR in favor of the consolidated version here:  
➡️ **#329 – Finalize Dart Sass migration & drop direct dependency**

That PR targets `release/5.0.0` and includes all changes from this one, along with updated docs and migration instructions.  
Looking forward to your feedback there!
